PWGCF/FemtoDream/femtoDreamProducerReducedTask.cxx

```diff
@@ -133,16 +135,33 @@ struct femtoDreamProducerReducedTask {
   float getMagneticFieldTesla(uint64_t timestamp)
   {
     // TODO done only once (and not per run). Will be replaced by CCDBConfigurable
-    static o2::parameters::GRPObject* grpo = nullptr;
-    if (grpo == nullptr) {
-      grpo = ccdb->getForTimeStamp<o2::parameters::GRPObject>("GLO/GRP/GRP", timestamp);
+    float output = -999;
+
+    if (ConfIsRun3 && !ConfIsMC) {
+      static o2::parameters::GRPMagField* grpo = nullptr;
+      if (grpo == nullptr) {
+        grpo = ccdb->getForTimeStamp<o2::parameters::GRPMagField>("GLO/Config/GRPMagField", timestamp);
+        if (grpo == nullptr) {
+          LOGF(fatal, "GRP object not found for timestamp %llu", timestamp);
+          return 0;
+        }
+        LOGF(info, "Retrieved GRP for timestamp %llu with magnetic field of %d kG", timestamp, grpo->getL3Current());
+      }
+      output = 0.1 * (grpo->getL3Current());
+    }
+
+    if (!ConfIsRun3 || (ConfIsRun3 && ConfIsMC)) {
+      static o2::parameters::GRPObject* grpo = nullptr;
       if (grpo == nullptr) {
-        LOGF(fatal, "GRP object not found for timestamp %llu", timestamp);
-        return 0;
+        grpo = ccdb->getForTimeStamp<o2::parameters::GRPObject>("GLO/GRP/GRP", timestamp);
+        if (grpo == nullptr) {
```

Comment:
Let's have it as it is now until we confirm the simulation presence on the new endpoint